lib/Common/Memory/LargeHeapBlock.cpp

```diff
@@ -979,9 +1057,18 @@ LargeHeapBlock::RescanOnePage(Recycler * recycler)
         RECYCLER_STATS_INC(recycler, markData.rescanLargeObjectCount);
         RECYCLER_STATS_ADD(recycler, markData.rescanLargeByteCount, header->objectSize);
 
-        if (!recycler->AddMark(objectAddress, header->objectSize))
+        size_t objectSize = header->objectSize;
+        if (this->InPageHeapMode())
         {
-            this->SetNeedOOMRescan(recycler);
+            // trim off the trailing part which is not a pointer
+            objectSize = objectSize - (objectSize%sizeof(void*));
```

Comment:
done

---
In reply to: [60996139](https://github.com/Microsoft/ChakraCore/pull/711#discussion_r60996139) [](ancestors = 60996139)